test/ttexalens/unit_tests/test_search.py

```
@@ -0,0 +1,42 @@
+# SPDX-FileCopyrightText: © 2025 Tenstorrent AI ULC
+
+# SPDX-License-Identifier: Apache-2.0
+
+# Unit test for ttexalens/util.py:search.
+import unittest
+from ttexalens.util import search
+from fnmatch import fnmatch
+
+class TestSearch(unittest.TestCase):
```

You should add some test cases that trigger exceptions. For example int(n) should raise an exception if a string that is not a number is passed as `n`. You can check for exceptions with `assertRaises`.